base/src/org/compiere/model/MOrder.java

```diff
@@ -1501,6 +1501,42 @@ public boolean isBinding() {
 		return binding;
 	}
 
+	/**
+	 * Force same Warehouse for Line but SO/PO
+	 * @param orderLine Order Line
+	 */
+	private void forceSameWarehouseToLine(int headerWarehouseId, MOrderLine orderLine) {
+		//	enforce WH
+		if (headerWarehouseId != 0)	{
+			if (headerWarehouseId != orderLine.getM_Warehouse_ID()) {
+				orderLine.setM_Warehouse_ID(headerWarehouseId);
+				orderLine.saveEx();
+			}
+			if (getAD_Org_ID() != orderLine.getAD_Org_ID()) {
+				orderLine.setAD_Org_ID(getAD_Org_ID());
+				orderLine.saveEx();
+			}
+		}
+	}
+
+	/**
+	 * Calculate Sizes
+	 * @param orderLines Order Lines
+	 */
+	private void calculateOrderSizes(List<MOrderLine> orderLines) {
+		List<BigDecimal> volumenList = new ArrayList<>();
+		List<BigDecimal> weightList = new ArrayList<>();
+		orderLines.forEach(orderLine -> {
```

Comment:
I don't know if is better use set for bigDecimal using a AtomiReference and add inside fiort forEach instead make two loops affter it

Reply:
Here a suggestion of code more clean without a unnecessary loop:

```Java
private void calculateOrderSizes(List<MOrderLine> orderLines) {
		setVolume(orderLines.stream().filter(orderLine -> orderLine.getM_Product_ID() > 0).map(orderLine -> {
			return Optional.ofNullable(orderLine.getProduct().getVolume()).orElse(Env.ZERO).multiply(Optional.ofNullable(orderLine.getQtyOrdered()).orElse(Env.ZERO));
		}).reduce(BigDecimal.ZERO, BigDecimal::add));
		setVolume(orderLines.stream().filter(orderLine -> orderLine.getM_Product_ID() > 0).map(orderLine -> {
			return Optional.ofNullable(orderLine.getProduct().getWeight()).orElse(Env.ZERO).multiply(Optional.ofNullable(orderLine.getQtyOrdered()).orElse(Env.ZERO));
		}).reduce(BigDecimal.ZERO, BigDecimal::add));
	}
```